bin/enrichEBEyeXMLDump.py

```diff
@@ -0,0 +1,166 @@
+    True
+    >>> diseases
+    set()
+    """
+    diseases, tissues, crossRefs  = (set([]), set([]), set([]))
+    for row in condensedSdrfStr.split("\n"):
+        arr = row.strip().split("\t")
+        if len(arr) > 4 and arr[3] == "factor":
+            if arr[4].lower() == "organism part":
+                tissues.add(arr[5].strip())
+                if len(arr) > 6:
+                    crossRefs.add(arr[6].split("/")[-1].strip())             
+            elif arr[4].lower() == "disease":
+                diseases.add(arr[5].strip())
+                if len(arr) > 6:
+                    crossRefs.add(arr[6].split("/")[-1].strip())
+    return (diseases, tissues, crossRefs)
+
+"""
+Add annotations in diseases, tissues, crossRefs to entry
+"""
+def addSampleAnnotationsToEntry(doc, entry, diseases, tissues, crossRefs):
+    """
+    >>> doc = minidom.Document()
+    >>> entry = doc.createElement('entry')
+    >>> addSampleAnnotationsToEntry(doc, entry, {'bladder carcinoma'}, {}, {'EFO_0000292'})
+    >>> entry.getElementsByTagName("additional_fields").length
+    1
+    >>> entry.getElementsByTagName("cross_references").length
+    1
+    >>> entry.getElementsByTagName("additional_fields").item(0).firstChild.toprettyxml(indent="").strip() == '<field name="disease">bladder carcinoma</field>'
+    True
+    >>> entry.getElementsByTagName("cross_references").item(0).firstChild.toprettyxml(indent="").strip() == '<ref dbName="efo" dbkey="EFO_0000292"/>'
+    True
+    >>> entry = doc.createElement('entry')
+    >>> addSampleAnnotationsToEntry(doc, entry, {}, {'adrenal'}, {})
+    >>> entry.getElementsByTagName("additional_fields").item(0).firstChild.toprettyxml(indent="").strip() == '<field name="tissue">adrenal</field>'
+    True
+    """
+    if diseases or tissues:
+        if len(entry.getElementsByTagName("additional_fields")) == 0:
+            createAppendElement(doc, entry, "additional_fields","")
+        additionalFields = entry.getElementsByTagName("additional_fields")[0]
+        if len(entry.getElementsByTagName("cross_references")) == 0:
+            createAppendElement(doc, entry, "cross_references","")
+        crossReferences = entry.getElementsByTagName("cross_references")[0]
+        for tissue in tissues:
+            createAppendElement(doc, additionalFields, 'field', tissue, [('name','tissue')])
+        for disease in diseases:
+            createAppendElement(doc, additionalFields, 'field', disease, [('name','disease')])
+        for ontologyTerm in crossRefs:
+            ontologyName = ontologyTerm.split("_")[0].lower()
+            createAppendElement(doc, crossReferences, 'ref', None, [('dbkey', ontologyTerm),('dbName', ontologyName)])
+
+condensedSDRFRootDir = None
+if __name__ == "__main__":
+    # Capture call arguments
+    if len(sys.argv) < 2:
+        print('Call argument needed, e.g. : ')
+        print(sys.argv[0] + ' ebeye_baseline_experiments.xml /path/to/condensed/sdrf')
```

Comment:
I would suggest in general to use argparse instead of positional arguments, then can get very quickly out of hand,  and argparse produces some help straightaway.

Reply:
Have added argparse, as requested. Thanks Pablo.